File: TransCoder_tokenized_test_set_functions/python/CHECK_LARGE_NUMBER_DIVISIBLE_6_NOT.py
def check ( st ) :
    n = len ( st )
    if ( ( ( int ) ( st [ n - 1 ] ) % 2 ) != 0 ) :
        return False
    digitSum = 0
    for i in range ( 0, n ) :
        digitSum = digitSum + ( int ) ( st [ i ] )
    return ( digitSum % 3 == 0 )

File: TransCoder_tokenized_test_set_functions/java/CHECK_LARGE_NUMBER_DIVISIBLE_6_NOT.java
static boolean check ( String str ) {
  int n = str . length ( );
  if ( ( str . charAt ( n - 1 ) - '0' ) % 2 != 0 ) return false;
  int digitSum = 0;
  for ( int i = 0;
  i < n;
  i ++ ) digitSum += ( str . charAt ( i ) - '0' );
  return ( digitSum % 3 == 0 );
}